src/cart/crt_iv.c

```diff
@@ -2502,17 +2517,32 @@ handle_ivupdate_response(const struct crt_cb_info *cb_info)
 	struct crt_iv_ops	 *iv_ops;
 	int			 rc;
 
+	iv_ops = crt_iv_ops_get(iv_info->uci_ivns_internal,
+				iv_info->uci_class_id);
+	D_ASSERT(iv_ops != NULL);
+
 	/* For bi-directional updates, transfer data back to child */
 	if (iv_info->uci_sync_type.ivs_flags & CRT_IV_SYNC_BIDIRECTIONAL) {
-		transfer_back_to_child(&input->ivu_key, iv_info, true,
-				       cb_info->cci_rc ?: output->rc);
+		if (iv_info->uci_bulk_hdl != CRT_BULK_NULL)
+			iv_ops->ivo_on_refresh(iv_info->uci_ivns_internal, &input->ivu_key, 0,
+					       &iv_info->uci_iv_value, false,
+					       cb_info->cci_rc ?: output->rc,
+					       iv_info->uci_user_priv);
+		else
+			iv_ops->ivo_on_refresh(iv_info->uci_ivns_internal, &input->ivu_key, 0,
+					       &output->ivo_iv_sgl, false,
+					       cb_info->cci_rc ?: output->rc,
+					       iv_info->uci_user_priv);
```

Comment:
why not combine as 
`		iv_ops->ivo_on_refresh(iv_info->uci_ivns_internal, &input->ivu_key, 0,
					       iv_info->uci_bulk_hdl ? &iv_info->uci_iv_value : &output->ivo_iv_sgl, 
					       false,
					       cb_info->cci_rc ?: output->rc,
					       iv_info->uci_user_priv);`

Reply:
sure